app/src/main/java/com/arctouch/codechallenge/util/Utils.kt

```diff
@@ -0,0 +1,17 @@
+package com.arctouch.codechallenge.util
+
+import android.content.Context
+import android.net.ConnectivityManager
+import com.arctouch.codechallenge.App
+import timber.log.Timber
+
+fun isDeviceHaveNoConnection(): Boolean {
+    return try {
+        val cm = App.instance.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val netInfo = cm.activeNetworkInfo
+        return netInfo == null || !netInfo.isConnectedOrConnecting
```

Comment:
Hello, i choose use this approach because, this is something that i know how to use, and when i build this project i didn't have the time to search a better approach.